api/server/fntracer.go

```diff
@@ -0,0 +1,130 @@
+package server
+
+import (
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/log"
+	"strings"
+)
+
+// FnTracer is a custom Tracer which wraps another another tracer
+// its main purpose is to wrap the underlying Span in a FnSpan,
+// which adds some extra behaviour required for sending tracing spans to prometheus
+type FnTracer struct {
+	wrappedTracer opentracing.Tracer
+}
+
+// NewFnTracer returns a new FnTracer which wraps the specified Tracer
+func NewFnTracer(tracerToWrap opentracing.Tracer) opentracing.Tracer {
+	newTracer := &FnTracer{}
+	newTracer.wrappedTracer = tracerToWrap
+	return newTracer
+}
+
+// FnTracer implements opentracing.Tracer
+func (thisFnTracer FnTracer) StartSpan(operationName string, opts ...opentracing.StartSpanOption) opentracing.Span {
```

Comment:
go nit: typically method receivers use short names, for this, `f`, `t`, or `ft` seem fitting.

Reply:
OK. Done.